controllers/addon/addon_controller.go

```diff
@@ -402,3 +402,57 @@ func (r *AddonReconciler) reconcile(ctx context.Context, addon *addonsv1alpha1.A
 
 	return ctrl.Result{}, nil
 }
+
+// Lists and filters pods with corev1.PodReasonUnschedulable status
+func (r *AddonReconciler) listUnschedulableAddonPods(
+	ctx context.Context,
+	addon *addonsv1alpha1.Addon,
+) (*corev1.PodList, error) {
+	addonPods := &corev1.PodList{}
+	unschedulablePods := &corev1.PodList{}
+
+	if addon.Status.Phase == addonsv1alpha1.PhaseReady {
+		return unschedulablePods, nil
+	}
+
+	if err := r.Client.List(
+		ctx,
+		addonPods,
+		client.InNamespace(addon.Spec.Install.OLMOwnNamespace.Namespace),
+	); err != nil {
+		return unschedulablePods, fmt.Errorf("failed listing addon pods: %w", err)
+	}
+
+	for _, pod := range addonPods.Items {
+		for _, podCond := range pod.Status.Conditions {
+			if podCond.Type == corev1.PodScheduled &&
+				podCond.Reason == corev1.PodReasonUnschedulable &&
+				podCond.Status == corev1.ConditionFalse {
+				unschedulablePods.Items = append(unschedulablePods.Items, pod)
+			}
+		}
+	}
+	return unschedulablePods, nil
+
+}
+
+// Gathers addon data for metric collection
+func (r *AddonReconciler) recordAddonMetrics(
+	ctx context.Context,
+	addon *addonsv1alpha1.Addon) (err error) {
+
+	if r.Recorder == nil {
+		return
+	}
+
+	unschedPods, err := r.listUnschedulableAddonPods(ctx, addon)
```

Comment:
With the current implementation, the reconciler checks for unschedulable addon pods on every reconcile. From a performance standpoint this isn't ideal, we should instead first check if an addon's 'Available' status is false, and if that's the case check for unschedulable events.

Reply:
That list function only proceeds if the addon's phase is not PhaseReady this is more proactive approach - happens before the actual issue. However I can change this to be more reactive if we only want this.

```
	if addon.Status.Phase == addonsv1alpha1.PhaseReady {
		return unschedulablePods, nil
	}

```

Reply:
I have updated this to use the avail condition instead of phase ready.